src/SoftDeletesBoolean.php

```diff
@@ -75,27 +81,31 @@ protected function performDeleteOnModel()
      */
     protected function runSoftDelete()
     {
-        $query = $this->newQueryWithoutScopes()->where($this->getKeyName(), $this->getKey());
+        $query = $$this->setKeysForSaveQuery($this->newModelQuery());
```

Comment:
```suggestion
        $query = $this->setKeysForSaveQuery($this->newModelQuery());
```

Reply:
Terrible code. Thank you for your review!